plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerLauncher.java

```diff
@@ -118,22 +134,22 @@ private void updateWorkspaceOnLSStarted() {
                 CompletableFuture.runAsync(
                     () -> projectSynchronizer.synchronize(registeredProject.getPath()));
               }
-              if (!registeredProject.getProblems().isEmpty()) {
-                try {
-
-                  projectManager.update(registeredProject);
-                  eventService.publish(new ProjectUpdatedEvent(registeredProject.getPath()));
-                } catch (ForbiddenException
-                    | ServerException
-                    | NotFoundException
-                    | ConflictException
-                    | BadRequestException e) {
-                  LOG.error(
-                      String.format(
-                          "Failed to update project '%s' configuration",
-                          registeredProject.getName()),
-                      e);
-                }
+              try {
+                projectManager.update(registeredProject);
+                eventService.publish(new ProjectClassPathChangedEvent(registeredProject.getPath()));
+                notifyTransmitter.sendNotification(
+                    new ExecuteCommandParams(
+                        Notifications.UPDATE_PROJECTS_CLASSPATH,
+                        Collections.singletonList(registeredProject.getPath())));
+              } catch (ForbiddenException
```

Comment:
Not there are other API exceptions that are not in the list.

Reply:
Can you be more clear on this? As I see all these exceptions extend `ApiException` so there is no need to catch them separately or is there?

Reply:
I'm curious which ones?

Reply:
Typehiearchy will answer that for you.

Reply:
As far as I see all possible exceptions that may be thrown here are: `ForbiddenException`, `ServerException`, `NotFoundException`,  `ConflictException` and `BadRequestException` - extends `ApiExceptions` so there is no reason to catch them separately, especially taking into account that you simply pass the exception to logger message.  In my opinion the statement 
> No, there are other API exceptions that are not in the list.

is senseless in this context, that is why I asked you to list the exceptions, in case I missed something by a mistake . However, your answer has indicated that you are not committed to a constructive dialogue, so I do not see the point to continue this controversy.

Reply:
By that logic, it would be ok to just catch Throwable, because all exception extend throwable. Catch should be as narrow as necessary.